Parser/asdl_c.py

```diff
@@ -782,28 +842,51 @@ def visitModule(self, mod):
 };
 
 static PyObject *
-make_type(const char *type, PyObject* base, const char* const* fields, int num_fields, const char *doc)
+make_type(
+    const char *type,
+    PyObject* base,
+    const char* const* fields,
+    const char* const* field_defaults,
+    int num_fields,
+    const char *doc
+)
 {
-    PyObject *fnames, *result;
+    PyObject *fnames, *fdefaults, *result;
     int i;
     fnames = PyTuple_New(num_fields);
-    if (!fnames) return NULL;
+    if (!fnames) {
+        return NULL;
+    }
+    fdefaults = PyTuple_New(num_fields);
+    if (!fdefaults) {
+        return NULL;
+    }
     for (i = 0; i < num_fields; i++) {
         PyObject *field = PyUnicode_InternFromString(fields[i]);
         if (!field) {
-            Py_DECREF(fnames);
-            return NULL;
+            goto cleanup;
         }
         PyTuple_SET_ITEM(fnames, i, field);
+        PyObject *field_default = PyUnicode_InternFromString(field_defaults[i]);
+        if (!field_default) {
+            goto cleanup;
+        }
+        PyTuple_SET_ITEM(fdefaults, i, field_default);
     }
-    result = PyObject_CallFunction((PyObject*)&PyType_Type, "s(O){OOOOOs}",
+    result = PyObject_CallFunction((PyObject*)&PyType_Type, "s(O){OOOOOOOs}",
                     type, base,
                     astmodulestate_global->_fields, fnames,
+                    astmodulestate_global->_field_defaults,
+                    fdefaults,
                     astmodulestate_global->__module__,
                     astmodulestate_global->_ast,
                     astmodulestate_global->__doc__, doc);
     Py_DECREF(fnames);
     return result;
+  cleanup:
```

Comment:
You are leaking `fdefaults` in the normal return case after `PyObject_CallFunction` is called, no? You can probably merge both exit modes.